src/App.tsx

```diff
@@ -15,47 +16,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
 export const App: React.FC = () => {
+  let copyGoods = [...goodsFromServer];
+  const [sortType, setsortType] = useState(SortType.NONE);
+  const [isReversed, setReversed] = useState(false);
+
+  copyGoods.sort((g1, g2) => {
```

Comment:
You can use useEffect here with sortType in the dependency array to sort the goods only when sortType changes.

Reply:
not fixed

Reply:
sorry, we had a conversation with previous checker, and solved to stay this points like its were.